sips/sips/sip-009-nft-standard.md

```diff
@@ -30,59 +30,110 @@ This SIP’s copyright is held by the Stacks Open Internet Foundation.
   )
 )
 ```
 
-## Implementation rules
+## Implementing in applications
+
+Developers who wish to interact with a non-fungible token contract should first be provided, or keep track of, various different non-fungible token implementations. When validating a non-fungible token contract, they should fetch the interface and/or source code for that contract. If the contract implements the trait, then the application can use this standard's contract interface for making transfers and getting other details defined in this standard.
+
+All of the functions in this trait return the `response` type, which is a requirement of trait definitions in Clarity. However, some of these functions should be "fail-proof", in the sense that they should never return an error. These "fail-proof" functions are those that have been recommended as read-only. If a contract that implements this trait returns an error for these functions, it may be an indication of a faulty contract, and consumers of those contracts should proceed with caution.
+
+## Use of native asset functions
+
+Although it is not possible to mandate in a Clarity trait, contract implementers should always use the built-in native assets that are provided as Clarity primitives. This allows clients to use Post Conditions (explained below), and takes advantages of other benefits, like native support for these asset balances and transfers through `stacks-blockchain-api`. The reference implementations included in this SIP use the native asset primitives, and provide a good boilerplate for their usage.
+
+The native asset primitives include:
+
+- `define-non-fungible-token`
+- `nft-burn?`
+- `nft-get-owner?`
+- `nft-mint?`
+- `nft-transfer?`
 
-1. Contracts must use a least one NFT asset. A post condition with deny mode and without any NFT condition about a changed owner must fail for `transfer` function calls.
-1. After a successful call to function `transfer`, the function `get-owner` must return the recipient of the `transfer` call as the new owner.
-1. If a call to function `get-owner` returns some principal `A` value then it must return the same value until `transfer` is called with principal `A` as a sender
-1. For any call to `get-owner`, resp. `transfer` with an id greater than `last-token-id`, the call should return a response `none`, resp. failed transfer. 
-1. The following error codes are defined
+## Use of post conditions
 
-| function | error | description |
-|----------|-------|-------------| 
-|`transfer`|`{kind: "nft-transfer-failed", code: from-nft-transfer}`| Error if the call failed due to the underlying asset transfer. The code `from-nft-transfer` is the error code from the native asset transfer function|
+In addition to built-in methods for non-fungible token contracts, the Stacks blockchain includes a feature known as Post Conditions. By defining post conditions, users can create transactions that include pre-defined guarantees about what might happen in that contract.
```

Comment:
This should not be part of the standard

Reply:
I was trying to accommodate this implementation rule you listed: "Contracts must use a least one NFT asset. A post-condition with deny mode and without any NFT condition about a changed owner must fail for transfer function calls." Isn't it not a post condition? should I rephrase it or remove it?